### PR TITLE
Compatibility with Rails 5.1

### PR DESCRIPTION
**Pull request by pschambacher**
/cc @zendesk/zendesk-rails-upgraders 

Only remarkable change: calling `establish_connection` will now automatically call the Resolver so we don't need to do it.

### CODE REVIEW

File: lib/active_record_shards/connection_switcher-5-0.rb [hunk 1]
```diff
@@ -20,9 +20,15 @@ def ensure_shard_connection
 
       pool = connection_handler.retrieve_connection_pool(spec_name)
       if pool.nil?
-        resolver = ActiveRecord::ConnectionAdapters::ConnectionSpecification::Resolver.new configurations
-        spec = resolver.spec(spec_name.to_sym, spec_name)
-        connection_handler.establish_connection spec
+        spec = if ActiveRecord::VERSION::MINOR.zero?
```
Comment (grosser) [inline lib/active_record_shards/connection_switcher-5-0.rb]: bah this indent ... how about:

```
spec =
  if dsfsf
    sdfdsf
  end
```

Comment (grosser) [inline lib/active_record_shards/connection_switcher-5-0.rb]: also this is kind of wrong ... would also work on 4.0 etc ... so do if MAJOR == 5 && MINOR == 0 ?

Comment (pschambacher) [inline lib/active_record_shards/connection_switcher-5-0.rb]: We're already in the `connection_switcher-5-0` file?

**Comment by grosser** [comment]
ah nvm

File: lib/active_record_shards/connection_switcher-5-0.rb [hunk 2]
```diff
@@ -20,9 +20,16 @@ def ensure_shard_connection
 
       pool = connection_handler.retrieve_connection_pool(spec_name)
       if pool.nil?
-        resolver = ActiveRecord::ConnectionAdapters::ConnectionSpecification::Resolver.new configurations
-        spec = resolver.spec(spec_name.to_sym, spec_name)
-        connection_handler.establish_connection spec
+        spec =
+          if ActiveRecord::VERSION::MAJOR == 5 && ActiveRecord::VERSION::MINOR.zero?
```
Comment (bquorning) [inline lib/active_record_shards/connection_switcher-5-0.rb]: No need to check `MAJOR`.

File: lib/active_record_shards/connection_switcher.rb [hunk 3]
```diff
@@ -190,7 +190,7 @@ def method_missing(method, *args, &block) # rubocop:disable Style/MethodMissing
 case "#{ActiveRecord::VERSION::MAJOR}.#{ActiveRecord::VERSION::MINOR}"
 when '3.2', '4.0', '4.1', '4.2'
   require 'active_record_shards/connection_switcher-4-0'
-when '5.0'
+when '5.0', '5.1'
   require 'active_record_shards/connection_switcher-5-0'
```
Comment (bquorning) [inline lib/active_record_shards/connection_switcher.rb]: We could make a separate file for 5.1.

File: .travis.yml [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-  - 2.2.5
-  - 2.3.1
+  - 2.1.10
+  - 2.2.4
+  - 2.3.3
+  - 2.4.0
 gemfile:
   - gemfiles/rails3.2.gemfile
-  - gemfiles/rails4.0.gemfile
-  - gemfiles/rails4.1.gemfile
```
Comment (bquorning) [inline .travis.yml]: I don't get this. We stop testing against Rails 4.0 and 4.1, but we keep all the version checks in the code. How do we know it works if we don't run the tests?

**Comment by grosser** [comment]
remove 4.0 / 4.1 ...

**Comment by bquorning** [comment]
> remove 4.0 / 4.1 ...

Yeah, but that's a separate discussion from adding 5.1 support.

**Pull request merged**